### PR TITLE
fix(@angular/cli): logic to determine if the installed CLI is out of …

### PR DESCRIPTION
**Pull request by alan-agius4**
With this change we now check if the current CLI version is the latest published version. If it is not, we install a temporary version to run the `ng update` with.

### CODE REVIEW

File: packages/angular/cli/commands/update-impl.ts [hunk 1]
```diff
@@ -190,6 +194,21 @@ export class UpdateCommand extends Command<UpdateCommandSchema> {
 
   // tslint:disable-next-line:no-big-function
   async run(options: UpdateCommandSchema & Arguments) {
+    // Check if the current installed CLI version is older than the latest version.
+    if (await this.checkCLILatestVersion(options.verbose)) {
+      this.logger.warn(
+        'The installed Angular CLI version is older then the latest published version.\n' +
```
Comment (clydin) [inline packages/angular/cli/commands/update-impl.ts]:
```suggestion
        'The installed Angular CLI version is older than the latest published version.\n' +
```

File: packages/angular/cli/tasks/install-package.ts [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+function getPackageManagerArguments(packageManager: PackageManager): PackageManagerOptions {
+  return packageManager === PackageManager.Yarn
+    ? {
+      silent: '--silent',
+      saveDev: '--dev',
+      install: 'add',
+      prefix: '--global-folder',
+    }
+    : {
+      silent: '--quite',
```
Comment (clydin) [inline packages/angular/cli/tasks/install-package.ts]:
```suggestion
      silent: '--quiet',
```

File: packages/angular/cli/tasks/install-package.ts [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    // Global installs on Unix systems go to {prefix}/lib/node_modules.
+    // Global installs on Windows go to {prefix}/node_modules (that is, no lib folder.)
+    tempNodeModules = join(tempPath, 'lib', 'node_modules');
+  } else {
+    tempNodeModules = join(tempPath, 'node_modules');
+  }
+
+  // Needed to resolve schematics from this location since we use a custom
+  // resolve strategy in '@angular/devkit-core/node'
+  process.env.NG_TEMP_MODULES_DIR = tempNodeModules;
```
Comment (clydin) [inline packages/angular/cli/tasks/install-package.ts]: The schematics resolution is the last place the custom resolver is being used.  We should move it to native methods when possible.  Can you add a note that this won't be needed when that refactor is done?

Comment (alan-agius4) [reply packages/angular/cli/tasks/install-package.ts]: will do, I actually have a following PR to remove that logic. Will try to land it in the coming days.

File: tests/legacy-cli/e2e/tests/i18n/ivy-localize-dl.ts [hunk 4]
```diff
@@ -19,11 +19,12 @@ export default async function() {
...
   }
 
-  let localizeVersion = '@angular/localize@' + readNgVersion();
+  let localizeVersion = readNgVersion();
   if (getGlobalVariable('argv')['ng-snapshots']) {
     localizeVersion = require('../../ng-snapshot/package.json').dependencies['@angular/localize'];
   }
-  await npm('install', `${localizeVersion}`);
+  await npm('install', `@angular/localize@${localizeVersion}`);
+  await npm('run', 'webdriver-update');
```
[the remark this answers is not in the document]
Comment (alan-agius4) [reply tests/legacy-cli/e2e/tests/i18n/ivy-localize-dl.ts]: Yeah, I but without it seemed to fail a lot.

**Pull request merged**